RFD 0201 - Automatic review of resource requests

bernardjkim opened the pull request:
Supports: https://github.com/gravitational/teleport/issues/51682
This PR updates the automatic review RFD with details of resource based requests.
[Rendered](https://github.com/gravitational/teleport/blob/78b58bdbbbfab6e6ca24b34128b5f94802e96198/rfd/0201-native-auto-approval.md)

bernardjkim committed:
Update RFD 0201

Updates the automatic review RFD with details of resource based
requests.

r0mant reviewed (commented):
I think we should support resource label based reviews rather than individual resources like you mention in the RFD.

bernardjkim committed:
Match by label instead of ID

bernardjkim committed:
Add a few more predicate expressions

- Replace has_labels with all_has_labels and some_has_labels.
- Add access_request.spec.requested_resources.length.

bernardjkim commented:
Hey @roraback, do you have any input on the WebUI changes proposed?

The changes might look something like this:

<img width="1267" alt="Screenshot 2025-06-27 at 9 35 35 AM" src="https://github.com/user-attachments/assets/171cb37e-ac61-4ad3-893d-7d0056b7e4e8" />

bernardjkim committed:
Use union and intersection approach

bernardjkim committed:
Merge branch 'master' into bernard/update-automatic-review-rfd

roraback reviewed (approved):
Sorry for the delayed review! This fell off my radar amidst onboarding new team members and Q3 planning. Looks quite robust and I appreciate that user stories are toward the top of the RFD! ❤️
